src/crate/client/sqlalchemy/compiler.py

```diff
@@ -107,7 +107,10 @@ class CrateDDLCompiler(compiler.DDLCompiler):
     def get_column_specification(self, column, **kwargs):
         colspec = self.preparer.format_column(column) + " " + \
             self.dialect.type_compiler.process(column.type)
-        # TODO: once supported add default here
+
+        default = self.get_column_default_string(column)
+        if default is not None:
```

Comment:
Do we intentionally want to intentionally ignore empty values like `[]`, `""`, `{}`.  Why is the reason behind not using?
```suggestion
        if default:
```
Trying to understand, thanks! 😁

Reply:
@sanchezfdezjavier 
Not sure if I correctly understand your questions, any value beside `None` will evaluate to `True`.
```
'' is not None => True
[] is not None => True
{} is not None => True
False is not None => True
```

While using `if default:` it would evaluate to `False` for all the above examples, see also https://docs.python.org/3/library/stdtypes.html#truth.

Reply:
Hi Javi,

`get_column_default_string()` probably returns a string, so in the case of non-empty strings, your suggestion might also work in this case.

But a) @seut is generally right, and b) explicitly testing against `is not None` also feels safer for me, because it does not suffer from any implicit bool-casting flaws.

With kind regards,
Andreas.

Reply:
Awesome, that makes sense. Thanks for clarifying!